### PR TITLE
Arbitrary file widget

### LINKED ISSUE #2135: Add arbitrary file question
As discussed in https://forum.opendatakit.org/t/question-type-file-not-image-video-or-audio/12558

Currently Javarosa sets the question control type to `CONTROL_FILE_CAPTURE` when the media type isn't recognized. This isn't completely ideal because it means a form developer can't set what file types they want to accept but I think it's ok as a start, especially before https://github.com/XLSForm/pyxform/issues/194 is addressed.

`ACTION_OPEN_DOCUMENT ` seems like the right approach for API 19+. I'm not sure what the best approach is for <19. `ACTION_GET_CONTENT`?

### PR DESCRIPTION
**Pull request by grzesiek2010**
Closes #2135 

#### What has been done to verify that this works as intended?
I've tested the widget adding various file types and sending them to aggregate.

#### Are there any risks to merging this code? If so, what are they?
It doesn't affect the rest of the code so I would say it's safe.

#### Do we need any specific form for testing your changes? If so, please attach one.
[arbitraryWidget.xml.txt](https://github.com/opendatakit/collect/files/1941859/arbitraryWidget.xml.txt)

#### Does this change require updates to documentation? If so, please file an issue at https://github.com/opendatakit/docs/issues/new and include the link below.

https://github.com/opendatakit/docs/issues/667

### CODE REVIEW

**Comment by grzesiek2010** [comment]
@lognaturel 
I tested some most popular file formats. Audio/Video/Image work well of course but also PDF, TXT
https://streamable.com/8q7yg

CSV and XLSX can't be opened but are downloaded as a file named `binaryData`.

**Comment by dcbriccetti** [comment]
I tested with a Jupyter Notebook (that should be sufficiently arbitrary). Aggregate gave it to me as `binaryData` (as you said happened for CSV above). What would it take to get the real name here?

File: collect_app/src/main/java/org/odk/collect/android/widgets/ArbitraryFileWidget.java [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+import org.odk.collect.android.utilities.FileUtil;
+import org.odk.collect.android.utilities.MediaManager;
+import org.odk.collect.android.utilities.MediaUtil;
+import org.odk.collect.android.widgets.interfaces.FileWidget;
+
+import java.io.File;
+
+import timber.log.Timber;
+
+public class ArbitraryFileWidget extends QuestionWidget implements FileWidget {
```
Comment (dcbriccetti) [inline collect_app/src/main/java/org/odk/collect/android/widgets/ArbitraryFileWidget.java:46]: For this class, did you write the code from scratch, or copy another class and modify it? (I’m just watching out for code duplication.)

**Review by dcbriccetti — APPROVED** [review]

File: collect_app/src/main/java/org/odk/collect/android/widgets/ArbitraryFileWidget.java (hunk 1 above)
Comment (grzesiek2010) [reply collect_app/src/main/java/org/odk/collect/android/widgets/ArbitraryFileWidget.java:46]: hmmm I'm not sure :smile: both options are possible but I would say from the scratch. I investigated duplications too and I think the code is ok.

**Comment by grzesiek2010** [comment]
I filed an issue about not supported file types https://github.com/opendatakit/aggregate/issues/241

**Comment by mmarciniak90** [comment]
**Tested with success**

Verified on Android 4.1, 4.2, 4.4, 5.1(not all cases because of file manager restrictions), 6.0, 7.0
Verified cases:
* Audio files: amr, ogg, m4a, mp3
* Video files: 3gp
* Image files: gif, jpeg, jpg, png 
* Doc files: pdf, doc, docx, txt, xls, xlsx, csv, xml, rtf
* Archive file format: zip, rar

@opendatakit-bot label "behavior verified"
@opendatakit-bot unlabel "needs testing"

**Comment by lognaturel** [comment]
🎉

**Pull request merged**